cooldowns/cooldown.py

```diff
@@ -438,3 +439,227 @@ def bucket(self) -> CooldownBucketProtocol:
     def func(self) -> Optional[Callable]:
         """Returns the wrapped function."""
         return self._func
+
+
+class TriggerCooldown:
+    """
+    Creates a trigger cooldown.
+
+    This is useful if you want to be able to trigger a specific time_period cooldown
+    inside the command itself.
+
+    TriggerCooldown creates two cooldonws in one instance:
+
+    - Normal cooldown. The same cooldown as @cooldowns.cooldown()
+    - Trigger cooldown. A secondary cooldown that can only be activate
+    with `.trigger()`
+
+    Parameters
+    ----------
+    limit : `int`
+        How many call's can be made in the time
+        period specified by ``time_period``.
+
+    time_period : `Union[float, datetime.timedelta]`
+        The time period related to ``limit``. This is seconds.
+
+    bucket : `Optional[CooldownBucketProtocol], optional`
+        The :class:`Bucket` implementation to use
+        as a bucket to separate cooldown buckets.
+
+    check : `Optional[MaybeCoro], optional`
+        A Callable which dictates whether
+        to apply the cooldown on current invoke.
+
+        If this Callable returns a truthy value,
+        then the cooldown will be used for the current call.
+
+        I.e. If you wished to bypass cooldowns, you
+        would return False if you invoked the Callable.
+    
+    cooldown_id: Optional[Union[int, str]]
+        Useful for resetting individual stacked cooldowns.
+        This should be unique globally,
+        behaviour is not guaranteed if not unique.
+
+        .. note::
+
+            This check will be given the same arguments as
+            the item you are applying the cooldown to.
+    
+    Usage
+    -----
+        - First create an instance of TriggerCooldown() with
+        the desired parameters.
+
+        ```
+        trigger_cooldown = cooldowns.TriggerCooldown(1, 5, cooldowns.SlashBucket.author)
+        ```
+        
+        - Then add the instance as a decorator to your command!
+
+        ```
+        @nextcord.slash_command()
+        @trigger_cooldown
+        async def command():
+        ```
+
+            The instance has to be defined in the same scope as the decorator!
+            Now, `command()` has applied a normal cooldown of `1 limit` and
+            `5 time_period`, as we defined it.
+
+        - Finally, inside your command, you can `trigger` the trigger cooldown:
+
+        ```
+        async def command():
+            # Do things
+            trigger_cooldown.trigger(30)
+            # You can still do things after this.
+            # Even you can `interaction.send()`.
+        ```
+
+            From the moment when the cooldown was triggered by `.trigger(30)`, every
+            single call to this command within 30 seconds will raise CallableOnCooldown!
```

Comment:
This is better suited to an examples entry in the documentation or in the decorators docstring

Reply:
I totally understand the reasons you may want the "ability to use these cooldowns directly via a decorator such as `@trigger_cooldown(...)` as well as via shared cooldowns" But here is a problem. 

To trigger a cooldown, you need an instance of it. The only way I can figure out doing this is by creating an instance of the trigger cooldown and then using it as a decorator.

 As far as I know, it is not possible to create a **direct decorator** because there is no instance to "trigger". When you decorate a function with, for example, `@trigger_cooldown(...)` you are just wrapping that function with the cooldown. The same happens with shared_cooldowns. Trigger cooldowns have their own behavior, they have to be an object, an instance, to be callable and modifiable. A trigger cooldown does not just "wrap" a function, but also activates when the function dictates to activate. 

The same happens with shared cooldowns, if we want to make trigger cooldowns shareable, then a new "sharing" method has to be created so that triggering is matched with the function that triggers.

My opinion is that one thing is just a trigger cooldown, and another thing is implementing methods like sharing and others.

So, the conclusion is that, as no direct cooldown decorator can be made, no docstring directly to the decorator can be created, because the decorator is an instance, a variable.

Reply:
True, however, when storing against an id someone could use the utility objects to fetch the cooldown instance via id as and where required for this functionality.

I would still like to see decorators, but with those comments in mind you may just need to make the cooldown id enforced so they can be fetched from the global state everytime when using methods such as [this](https://function-cooldowns.readthedocs.io/en/latest/modules/util.html#cooldowns.get_cooldown)

Reply:
I will keep that in mind to search for a way to do it with `get_cooldown()`.

Reply:
I meant to have `get_cooldown` get the `TriggerCooldown` instance, not the underlying cooldowns which is what appears to occur now